generator/src/main/java/com/scottlogic/deg/generator/generation/string/generators/ChecksumStringGenerator.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2019 Scott Logic Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.scottlogic.deg.generator.generation.string.generators;
+
+import com.scottlogic.deg.common.ValidationException;
+import com.scottlogic.deg.generator.utils.RandomNumberGenerator;
+
+import java.util.stream.Stream;
+
+public class ChecksumStringGenerator implements StringGenerator {
+
+    private final StringGenerator checksumlessGenerator;
+    private final ChecksumMaker checksumMaker;
+
+    public ChecksumStringGenerator(StringGenerator checksumlessGenerator, ChecksumMaker checksumMaker) {
+        this.checksumlessGenerator = checksumlessGenerator;
+        this.checksumMaker = checksumMaker;
+    }
+
+    @Override
+    public Stream<String> generateAllValues() {
+        return checksumlessGenerator.generateAllValues()
+            .map(string -> string + checksumMaker.makeChecksum(string));
```

Comment:
Opinionated optional comment: Make these three `.map(string -> string + checksumMaker.makeChecksum(string))` into either a method that returns a `Function<String, String>`, or a private static final `Function<String, String>`.